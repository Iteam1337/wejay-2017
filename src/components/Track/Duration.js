// @flow

import React from 'react'
import { timeParser } from 'utils/parsers'
import styled from 'styled-components'

type DurationProps = {
  duration: number,
}

const DurationWrap = styled.div`
  color: rgba(54, 61, 67, 1);
  font-size: 14px;
  text-align: right;
`

const Duration = ({ duration }: DurationProps) => {
  return <DurationWrap>{timeParser(duration)}</DurationWrap>
}

export default Duration
