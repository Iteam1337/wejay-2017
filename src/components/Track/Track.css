.Track {
  align-items: center;
  color: #f0f0f0;
  display: flex;
  transition: opacity 300ms ease-in-out;
  opacity: 1; }
  .Track:not(:last-child) {
    margin-bottom: 20px; }

.Track__pending {
  opacity: 0.3; }

.Track__content {
  flex: 1;
  margin-left: 10px; }

.Track__artist {
  font-size: 14px;
  font-weight: 700;
  margin-bottom: 5px; }

.Track__name {
  font-size: 13px; }

.Track__duration {
  color: #666666; }
