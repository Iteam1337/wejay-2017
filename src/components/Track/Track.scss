.Track {
  align-items: center;
  border-radius: 0 4px 4px 0;
  display: grid;
  grid-column-gap: 20px;
  grid-template-columns: 80px 1fr 30px;
  transition: opacity 300ms ease-in-out, background-color 200ms ease-in-out;
  opacity: 1;
  padding-right: 20px;

  &:hover {
    background-color: rgba(255, 134, 0, 0.05);
  }
}

.Track.Track__pending {
  opacity: 0.3;

  .Track__artist {
    animation: 1.5s pending alternate-reverse infinite;
    background: #333;
    border-radius: 2px;
    height: 21px;
    transform-origin: 0;
    width: 80px;
  }

  .Track__name {
    animation: 2s pending alternate-reverse infinite;
    background: #999;
    border-radius: 2px;
    height: 19px;
    transform-origin: 0;
    width: 200px;
  }

  .Track__duration {
    background: #ccc;
    border-radius: 2px;
    height: 16px;
    width: 30px;
  }
}

@keyframes pending {
  from {
    transform: scaleX(0);
  }
  to {
    transform: scaleX(1);
  }
}

.Track__content {
  flex: 1;
  margin-left: 10px;
}

.Track__artist {
  color: #2a2a48;
  font-size: 18px;
  font-weight: 700;
  margin-bottom: 5px;
}

.Track__name {
  color: #666;
  font-size: 16px;
  margin-bottom: 5px;
}

.Track__duration {
  color: #999;
  font-size: 14px;
}

.Track__cover {
  position: relative;
  transition: transform 500ms ease-in-out;

  &:hover {
    transform: translateY(-2px);
  }
}

.Gravatar.Track__gravatar {
  bottom: -10px;
  position: absolute;
  right: -10px;
  z-index: 3;
}
