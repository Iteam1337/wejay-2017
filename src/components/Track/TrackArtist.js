// @flow

import * as WejayApi from '__generated__/types.flow'
import React from 'react'
import styled from 'styled-components'

type TrackArtistProps = {
  artists: $PropertyType<WejayApi.TrackInfoFragment, 'artists'>,
}

const TrackArtistItem = styled.div`
  color: rgba(54, 61, 67, 0.6);
  font-size: 12px;
`

const TrackArtist = ({ artists }: TrackArtistProps) => {
  return (
    <TrackArtistItem>{artists.map(t => t.name).join(', ')}</TrackArtistItem>
  )
}

export default TrackArtist
