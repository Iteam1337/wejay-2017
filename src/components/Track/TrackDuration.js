// @flow

import React from 'react'
import styled from 'styled-components'
import trackDuration from 'track-duration'
type TrackDurationProps = {
  duration: number,
}

const TrackDurationWrap = styled.div`
  color: rgba(54, 61, 67, 0.6);
  font-size: 12px;
  font-weight: 500;
  text-align: center;
`

const TrackDuration = ({ duration }: TrackDurationProps) => {
  return <TrackDurationWrap>{trackDuration.parse(duration)}</TrackDurationWrap>
}

export default TrackDuration
