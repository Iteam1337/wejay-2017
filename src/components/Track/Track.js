// @flow

import * as WejayApi from '__generated__/types.flow'
import React from 'react'
import Cover from 'components/Cover'
import Duration from './Duration'
import TrackArtist from './TrackArtist'
import styled from 'styled-components'

type TrackProps = {
  track: WejayApi.TrackInfoFragment,
}

const TrackRow = styled.div`
  align-items: center;
  border-bottom: 1px solid #eaecef;
  display: grid;
  grid-template-columns: 60px 1fr 60px;
  padding: 15px 20px;
`

const TrackMeta = styled.div`
  line-height: 1.4;
`

const TrackName = styled.div`
  color: rgba(54, 61, 67, 0.6);
  font-size: 12px;
`

const TrackItem = ({ track }: TrackProps) => {
  if (!track) {
    return null
  }

  return (
    <TrackRow>
      <Cover small track={track} width={40} />

      <TrackMeta>
        <TrackArtist artists={track.artists} />
        <TrackName>{track.name}</TrackName>
      </TrackMeta>

      <Duration duration={track.duration} />
    </TrackRow>
  )
}

export default TrackItem
