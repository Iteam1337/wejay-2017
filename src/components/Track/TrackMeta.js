// @flow

import * as WejayApi from '__generated__/types.flow'
import React from 'react'
import styled from 'styled-components'
import TrackArtist from './TrackArtist'

type TrackMetaProps = {
  artists: $PropertyType<WejayApi.TrackInfoFragment, 'artists'>,
  name: string,
}

export const TrackMetaWrap = styled.div`
  line-height: 1.4;
`

export const TrackName = styled.div`
  color: rgba(54, 61, 67, 1);
  font-size: 14px;
  font-weight: 500;
`

const TrackMeta = ({ artists, name }: TrackMetaProps) => {
  return (
    <TrackMetaWrap>
      <TrackName>{name}</TrackName>
      <TrackArtist artists={artists} />
    </TrackMetaWrap>
  )
}

export default TrackMeta
