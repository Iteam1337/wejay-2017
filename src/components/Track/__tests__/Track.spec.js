import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Track from '../Track'

describe('components/Track', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <Track
        track={{
          artists: [
            {
              name: 'artist',
            },
            { name: 'artist two' },
          ],
          duration: 200,
          name: 'track',
          user: {
            id: '1',
          },
        }}
      />
    )
  })

  it('returns null if no track', () => {
    component.setProps({
      track: undefined,
    })

    expect(component.type()).toBeNull()
  })

  it('renders a pending Track if no duration (optimisticResponse)', () => {
    component.setProps({
      track: {
        duration: 0,
        user: {
          id: '1',
        },
      },
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('renders Track', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
