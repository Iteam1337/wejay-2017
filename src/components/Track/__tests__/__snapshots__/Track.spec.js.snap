// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Track renders Track 1`] = `
<styled.div>
  <Component
    small={true}
    track={
      Object {
        "artists": Array [
          Object {
            "name": "artist",
          },
          Object {
            "name": "artist two",
          },
        ],
        "duration": 200,
        "name": "track",
        "user": Object {
          "id": "1",
        },
      }
    }
    width={40}
  />
  <TrackMeta
    artists={
      Array [
        Object {
          "name": "artist",
        },
        Object {
          "name": "artist two",
        },
      ]
    }
    name="track"
  />
  <TrackDuration
    duration={200}
  />
  <Gravatar
    email=""
    id="1"
    size={30}
  />
</styled.div>
`;

exports[`components/Track renders a pending Track if no duration (optimisticResponse) 1`] = `
<styled.div>
  <Component
    small={true}
    track={
      Object {
        "duration": 0,
        "user": Object {
          "id": "1",
        },
      }
    }
    width={40}
  />
  <TrackMeta />
  <TrackDuration
    duration={0}
  />
  <Gravatar
    email=""
    id="1"
    size={30}
  />
</styled.div>
`;
